classes/db/DAO.inc.php

```diff
@@ -160,6 +182,46 @@ function &retrieveLimit($sql, $params = false, $numRows = false, $offset = false
 		$start = Core::microtime();
 		$dataSource = $this->getDataSource();
 		$result = $dataSource->selectLimit($sql, $numRows === false ? -1 : $numRows, $offset === false ? -1 : $offset, $params !== false && !is_array($params) ? array($params) : $params);
+
+		if ($this->isSqlServerFakeError($dataSource)) {
+			return $result;
+		}
+
+		if ($dataSource->errorNo()) {
+			fatalError('DB Error: ' . $dataSource->errorMsg());
+		}
+		return $result;
+	}
+
+	/**
+	 * Execute a cached SELECT SQL statement with LIMIT on the rows returned.
+	 * @param $sql string the SQL statement
+	 * @param $params array parameters for the SQL statement
+	 * @param $numRows int maximum number of rows to return in the result set
+	 * @param $offset int row offset in the result set
+	 * @param $secsToCache int number of seconds
+	 * @return ADORecordSet
+	 */
+	function &retrieveLimitCached($sql, $params = false, $numRows = false, $offset = false, $secsToCache = 3600, $callHooks = true) {
+		if ($callHooks === true) {
+			$trace = debug_backtrace();
+			// Call hooks based on the calling entity, assuming
+			// this method is only called by a subclass. Results
+			// in hook calls named e.g. "sessiondao::_getsession"
+			// (all lowercase).
+			$value = null;
+			if (HookRegistry::call(strtolower_codesafe($trace[1]['class'] . '::_' . $trace[1]['function']), array(&$sql, &$params, &$numRows, &$offset, &$secsToCache, &$value))) {
+				return $value;
+			}
+		}
+
+		$dataSource = $this->getDataSource();
+		$result = $dataSource->selectLimit($sql, $numRows === false ? -1 : $numRows, $offset === false ? -1 : $offset, $params !== false && !is_array($params) ? array($params) : $params, $secsToCache);
```

Comment:
Are you sure this works? I don't see a 5th parameter to `selectLimit` documented here: http://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:selectlimit
I think [cacheSelectLimit](http://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:cacheselectlimit) might be what you're looking for.

Reply:
Yes, the called method (in file adodb-pdo.inc.php) looks like this :

```
	function SelectLimit($sql,$nrows=-1,$offset=-1,$inputarr=false,$secs2cache=0)
	{
		$save = $this->_driver->fetchMode;
		$this->_driver->fetchMode = $this->fetchMode;
		$this->_driver->debug = $this->debug;
		$ret = $this->_driver->SelectLimit($sql,$nrows,$offset,$inputarr,$secs2cache);
		$this->_driver->fetchMode = $save;
		return $ret;
	}
```

Reply:
Hmm, odd that it isn't documented. In that case, maybe it's best to just add the optional extra parameter to the `retrieveLimit` function, rather than introducing a new function that's essentially a duplicate.

Reply:
Not a bad idea! It is indeed the exact same method.